com.unity.shadergraph/Editor/Data/Graphs/SerializableTexture.cs

```diff
@@ -19,20 +25,26 @@ public Texture texture
         {
             get
             {
-                if (string.IsNullOrEmpty(m_SerializedTexture))
-                    return null;
-                var tex = new TextureHelper();
-                EditorJsonUtility.FromJsonOverwrite(m_SerializedTexture, tex);
-                return tex.texture;
+                if (!string.IsNullOrEmpty(m_SerializedTexture))
+                {
+                    var textureHelper = new TextureHelper();
+                    EditorJsonUtility.FromJsonOverwrite(m_SerializedTexture, textureHelper);
+                    m_SerializedTexture = null;
+                    m_Guid = AssetDatabase.AssetPathToGUID(AssetDatabase.GetAssetPath(textureHelper.texture));
+                    m_Texture = textureHelper.texture;
+                }
+                else if (!string.IsNullOrEmpty(m_Guid) && m_Texture == null)
+                {
```

Comment:
Here again :) curly ones

Reply:
Intentional, as the other block in the if above uses curly braces.